src/StreamingDataFrames/README.md

```diff
@@ -1,3 +1,258 @@
+
+
+<br><br>
+
+
+# Overview
+
+## What is Streaming DataFrames (SDF)?
+Streaming DataFrames is a Pandas-like Python application/microservice framework for 
+Kafka or Redpanda. 
+
+It uses a similar interface as Pandas for data transformation (ETL), along with 
+additional kafka-specific operations or features.
+
+<br>
+
+## How does it work?
+There are two primary components: a `StreamingDataFrame` (`dataframe`) and it's handler, 
+`Application` (`app`).
+
+Your `dataframe` is a pre-defined (declarative) pipeline that you give to your `app`
+to execute. All of your ETL is done here. 
+
+The `app` manages various Kafka-related setup/teardown and message lifecycle 
+(consuming, committing). It processes each message with the `dataframe` you provide it.
+
+<br>
+
+## Current Features
+
+- A simple framework with familiar Pandas interface to ease newcomers into streaming.
+- Easily integrates with other Python frameworks and ML models.
+- Support for basic serialization formats, including JSON (and Quix-specific)
+- Support for basic stateful operations using RocksDB
+- At-least-once Kafka processing guarantees
+- Designed to run and scale resiliently via container orchestration (like Kubernetes)
+- Also easily runs locally for development and debugging
+- Seamless integration with the Quix platform.
+
+Unfamiliar with the Quix platform? We make building and releasing your applications 
+easy by managing an entire infrastructure stack (Kafka, Kubernetes, CI/CD) for you 
+under the hood, with production software best practices in mind, and all backed by Git. 
+
+[Be sure to check us out at quix.io](https://quix.io/) to get up and running with 
+Kafka, fast!
+
+<br>
+
+## Upcoming Improvements
+
+- Stateful Windowing: tumbling, hopping, and sliding windows
+- Group-bys and joins (for merging topics/keys)
+- State recovery based on Kafka changelog topics
+- Support for exactly-once Kafka processing (aka transactions)
+- Other serialization support like Avro and Protobuf support
+- Schema Registry support
+- Multiple input topic support (for non-stateful applications)
+
+<br>
+
+## Want to get Involved?
+
+We strongly believe in the open source model and would love to build a strong 
+community with you, so please join us!
+
+While we are still working on out contribution guidelines, here are some ways you can 
+get involved right now:
+
+- [Join our community mailing list](https://share.hsforms.com/1WWc91mC_SqCvk8xeUWibMw4yjw2)
+- [Check out our project on GitHub](https://github.com/quixio/quix-streams/)
+- [Hang out with us on Slack](https://join.slack.com/t/stream-processing/shared_invite/zt-25lo3wxo7-28hlegDhR2XqyLWjBQynwA)
+
+
+<br><br>
+
+# The Basics
+
+Here's the minimum you need to know to get started with Streaming DataFrames.
+
+More detailed breakdowns can be found in the [documentation folder](./documentation).
+
+## The `Application` class
+
+This is the handler and entrypoint of any `quixstreams` application.
```

Comment:
I don't like the way we say '`quixstreams application`' because I immediately  get a question if I can use it without outside of Quix platform.

I also don't like the name `Application` but I don't have anything better to suggest atm. `Stream` came up to me as an idea but its probably bad as well